### PR TITLE
Replace TWTS by original well tested version

### PR DESCRIPTION
**Pull request by psychocoderHPC**
The current TWTS version includes some bugs, it is very hard to follow optimizations done in the past and identify the bug.
@steindev and I decided to start again from @BeyondEspresso well-tested version and apply optimizations step by step.

register footprint optimizations: 
 - FieldB: 95 -> 89 register
 - FieldE: 73 -> 60 register 

- [ ]  do not merge, @steindev will do the validation

### CODE REVIEW

**Comment by BeyondEspresso** [comment]
This PR is IMO a bad idea as this is the work, which I already did with PR #3439. There are a number of details that had to be taken into account during the implementation and likely could get lost here. Why did we not follow the strategy we agreed on yesterday?

File: include/picongpu/fields/background/templates/TWTS/BField.tpp [hunk 1]
```diff
@@ -715,66 +725,77 @@ namespace picongpu
...
-                       + complex_T(0, 4) * cspeed * om0 * t * wy2 * z2 * sinPhi
-                       + float_T(2.0) * cspeed * om02 * tauG2 * wy2 * z2 * sinPhi
-                       - complex_T(0, 2) * om0 * wy2 * z2 * z * sinPhi
-                       - float_T(4.0) * cspeed * om0 * t * wy2 * y * rho0 * tanPhi2
-                       + float_T(4.0) * om0 * wy2 * y * z * rho0 * tanPhi2
+                       - complex_T(0, 2) * t2 * z * c2_om0_wy2_roh0
+                       - float_T(2.0) * cspeed2 * om02_wy2_sinPhi * t * tauG2 * wy2 * z
+                       - complex_T(0, 2) * cspeed2 * om0_tauG2 * y2 * z * sinPhi
+                       + complex_T(0, 4) * c_t * z2 * om0_wy2_sinPhi
+                       + float_T(2.0) * cspeed * om02_wy2_sinPhi * tauG2 * wy2 * z2
```
Comment (psychocoderHPC) [inline include/picongpu/fields/background/templates/TWTS/BField.tpp:763]: one `wy2` to much

File: include/picongpu/fields/background/templates/TWTS/BField.tpp [hunk 2]
```diff
@@ -715,66 +725,77 @@ namespace picongpu
...
-                       - complex_T(0, 2) * cspeed2 * om0 * t2 * wy2 * z * sinPhi
-                       - float_T(2.0) * cspeed2 * om02 * t * tauG2 * wy2 * z * sinPhi
-                       - complex_T(0, 2) * cspeed2 * om0 * tauG2 * y2 * z * sinPhi
-                       + complex_T(0, 4) * cspeed * om0 * t * wy2 * z2 * sinPhi
-                       + float_T(2.0) * cspeed * om02 * tauG2 * wy2 * z2 * sinPhi
-                       - complex_T(0, 2) * om0 * wy2 * z2 * z * sinPhi
-                       - float_T(4.0) * cspeed * om0 * t * wy2 * y * rho0 * tanPhi2
-                       + float_T(4.0) * om0 * wy2 * y * z * rho0 * tanPhi2
+                       - complex_T(0, 2) * t2 * z * c2_om0_wy2_roh0
+                       - float_T(2.0) * cspeed2 * om02_wy2_sinPhi * t * tauG2 * wy2 * z
```
Comment (psychocoderHPC) [inline include/picongpu/fields/background/templates/TWTS/BField.tpp:760]: one `wy2` too much

File: include/picongpu/fields/background/templates/TWTS/BField.tpp [hunk 3]
```diff
@@ -715,66 +725,77 @@ namespace picongpu
...
                 const complex_T helpVar3
-                    = (-cspeed2 * k * om0 * tauG2 * wy2 * x2 - float_T(2.0) * cspeed2 * om0 * t2 * wy2 * rho0
-                       + complex_T(0, 2) * cspeed2 * om02 * t * tauG2 * wy2 * rho0
-                       - float_T(2.0) * cspeed2 * om0 * tauG2 * y2 * rho0
-                       + float_T(4.0) * cspeed * om0 * t * wy2 * z * rho0
-                       - complex_T(0, 2) * cspeed * om02 * tauG2 * wy2 * z * rho0
-                       - float_T(2.0) * om0 * wy2 * z2 * rho0
-                       - complex_T(0, 8) * om0 * wy2 * y * (cspeed * t - z) * z * sinPhi2_2
+                    = (-cspeed2 * k * om0_tauG2 * wy2 * x2 - float_T(2.0) * t2 * c2_om0_wy2_roh0
+                       + complex_T(0, 2) * t * tauG2 * c2_om0_wy2_roh0
```
Comment (steindev) [inline include/picongpu/fields/background/templates/TWTS/BField.tpp]: `om0` missing

File: include/picongpu/fields/background/templates/TWTS/BField.tpp [hunk 4]
```diff
@@ -715,66 +725,77 @@ namespace picongpu
...
+                       - float_T(2.0) * cspeed2 * om0_tauG2 * y2 * rho0
+                       + float_T(4.0) * c_t * z * om0_wy2_roh0
+                       - complex_T(0, 2) * cspeed * tauG2 * z * om0_wy2_roh0
```
Comment (steindev) [inline include/picongpu/fields/background/templates/TWTS/BField.tpp]: `om0` missing

File: include/picongpu/fields/background/templates/TWTS/BField.tpp [hunk 5]
```diff
@@ -715,66 +725,77 @@ namespace picongpu
...
                            * sinPhi2_4
-                       - complex_T(0, 2) * cspeed2 * om0 * t2 * wy2 * z * sinPhi
-                       - float_T(2.0) * cspeed2 * om02 * t * tauG2 * wy2 * z * sinPhi
-                       - complex_T(0, 2) * cspeed2 * om0 * tauG2 * y2 * z * sinPhi
-                       + complex_T(0, 4) * cspeed * om0 * t * wy2 * z2 * sinPhi
-                       + float_T(2.0) * cspeed * om02 * tauG2 * wy2 * z2 * sinPhi
-                       - complex_T(0, 2) * om0 * wy2 * z2 * z * sinPhi
-                       - float_T(4.0) * cspeed * om0 * t * wy2 * y * rho0 * tanPhi2
-                       + float_T(4.0) * om0 * wy2 * y * z * rho0 * tanPhi2
+                       - complex_T(0, 2) * t2 * z * c2_om0_wy2_roh0
```
Comment (steindev) [inline include/picongpu/fields/background/templates/TWTS/BField.tpp]: `rho0` instead of `sinPhi`

File: include/picongpu/fields/background/templates/TWTS/BField.tpp (hunk 2 above)
Comment (steindev) [reply include/picongpu/fields/background/templates/TWTS/BField.tpp:760]: `wy2` too much

**Comment by steindev** [comment]
Aaaarggh. Sorry @psychocoderHPC! I messed up your branch by trying to squash my fixes somewhere in between your commits. But this just lead to the fact, that my fixes where overwritten again by later commits. Unfortunately I just recognized this after pushing. Do **not** pull! Instead force push your original commits again. Sorry! :disappointed:

**Comment by steindev** [comment]
Ok. I had another version lying around. So I could fix it myself. It is also rebased to current dev. Next I will apply the fixes. In a seperate commit! :sweat_smile:

**Comment by psychocoderHPC** [comment]
I will close this PR since we can now use the incident field for TWTS.
